text-timeseries changes.

rma-rripken opened the pull request.

rma-rripken committed:
WIP changes.

rma-rripken committed:
Renamed controller and changed url.  Adding StandardTextTimeSeriesRow.

rma-rripken committed:
Merge remote-tracking branch 'origin/develop' into feature/text_time_series

rma-rripken committed:
More WIP.  Adding method to build TextTimeSeries from view.

rma-rripken committed:
Added property annotation so that jackson doesn't get confused.

rma-rripken committed:
Merge remote-tracking branch 'origin/feature/update_jooq_version' into feature/text_time_series

rma-rripken committed:
Renaname timeSeriesText package to timeseriestext. Added braces to intellij generated equals methods.

rma-rripken committed:
Added Controllers constants

rma-rripken committed:
Added parameters.

rma-rripken committed:
Added enum for DeleteMode

rma-rripken committed:
Changed method signature

rma-rripken committed:
Created placeholder IT.

rma-rripken committed:
renamed some methods

rma-rripken committed:
SQL resource files for reg text timeseries.

rma-rripken committed:
Fixed test

rma-rripken committed:
Merge remote-tracking branch 'origin/feature/update_jooq_version' into feature/text_time_series

rma-rripken committed:
Adding Std text timeseries IT.

rma-rripken committed:
Added an enum for which TimeSeriesText mode the user wants.

rma-rripken committed:
Renamed TextTimeSeries value fields.  Introduced an optional catalog of used ids for standard text timeseries.

rma-rripken committed:
Renamed timeseriestext package to texttimeseries.  Refactored the standard and regular dao a little to expose row retrieve methods.

rma-rripken committed:
Inlined a bunch of the std args.

rma-rripken committed:
Added std delete test.

rma-rripken committed:
Added regular delete test.

rma-rripken committed:
Cleanup minor issues

rma-rripken committed:
Removing debugging System.out from tests.

rma-rripken committed:
Delete cwms-data-api/src/test/java/cwms/cda/data/dao/TimeSeriesTextDaoTest.java

Removing test that wasn't meant to be added.

rma-rripken committed:
Merge remote-tracking branch 'origin/feature/text_time_series' into feature/text_time_series

rma-rripken committed:
Allow spaces in DeleteMethod lookup.

MikeNeilson commented on cwms-data-api/src/main/java/cwms/cda/data/dao/LocationLevelsDaoImpl.java:
```diff
@@ -587,7 +587,7 @@ public TimeSeries retrieveLocationLevelAsTimeSeries(ILocationLevelRef levelRef,
         ZTSV_ARRAY locLvlValues = CWMS_LEVEL_PACKAGE.call_RETRIEVE_LOC_LVL_VALUES3(dsl.configuration(),
                 specifiedTimes, locationLevelId, levelUnits, attributeId, attributeValue, attributeUnits,
-                "UTC", officeId);
+                "UTC", officeId, "VN");
```
what is VN?

rma-rripken replied:
I don't entirely remember making that change but it looks like it was related to updating the jooq codegen.  The method signature now has a p_level_precedence parameter and the procedure def lists "VN" as the default. 

Here's the doc on it:  
 @param p_level_precedence   One or two characters that specify the precedence of virtual and non-virtual location levels in the results.
 *                             The default value 'VN' gives virtual location levels higher precedence, allowing them to hide non-virtual location levels
 *                             where the two types overlap in time. Valid value are:
 * <ul>
 *   <li><b>N</b> specifies results from non-virtual (normal) location levels only
 *   <li><b>V</b> specifies results from virtual location levels only
 *   <li><b>NV</b> specifies results from non-virtual (normal) location levels where they exist, with virtual location levels allowed where non-virtual levels don't exist
 *   <li><b>VN</b> (default) specifies results from virtual location levels where they exist, with non-virtual location levels allowed where virtual levels don't exist
 * </ul>

rma-rripken replied:
I could change it to null - thats probably the more correct thing to do.

rma-rripken replied:
no, passing null doesn't seem to trigger the default.  not sure how to make the call from jooq without specifying the parameter so that the default will take effect.

MikeNeilson replied:
I'd leave it in just put a short comment about the behavior and where to find more information. I doubt I'm the only one that might get confused by that.

MikeNeilson commented on cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/RegularTimeSeriesTextDao.java:
```diff
@@ -0,0 +1,244 @@
+    @Nullable
+    private static List<RegularTextTimeSeriesRow> buildRows(ResultSet rs) throws SQLException {
+        OracleTypeMap.checkMetaData(rs.getMetaData(), timeSeriesTextColumnsList, TYPE);
+        List<RegularTextTimeSeriesRow> rows = null;
```
shouldn't this just allocate the array list or this there a specific reason we're allowing null if rs.next() is empty?

rma-rripken replied:
You are probably right.  I will revisit null vs empty.

MikeNeilson reviewed (commented):
Looks reasonable thus far.

NOTE: This review is mostly so I can use GH's "since last reviewed" feature.

Commit:
Added test for store

Commit:
Added test for store

MikeNeilson commented on cwms-data-api/src/test/java/cwms/cda/data/dao/texttimeseries/RegularTimeSeriesTextDaoTestIT.java:
```diff
@@ -210,4 +209,72 @@ private void testDelete(RegularTimeSeriesTextDao dao) {
+
+    @Test
+    void testStore() throws SQLException {
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
```
FYI the before all needs to create the location with the base case handler.

Wasn't able to run the test locally due to the location not already existing.

MikeNeilson reviewed (commented):
again, mostly set for GH tracking; but wasn't able to run the test locally due to one comment provided.


Oh and you might want to rebase from develop to pick up the gradle changes. (not a huge deal but it does solve a problem for me at least of always having to make sure I have the local environment set to the correct JDK.)

rma-rripken committed:
Merge remote-tracking branch 'origin/develop' into feature/text_time_series

rma-rripken committed:
Added a create_ts that also takes a utc_offset.

rma-rripken committed:
Refactored TextTS tests to use createLocation and createTs test methods.

rma-rripken commented:
Debugging text-timeseries pl/sql errors required https://bitbucket.hecdev.net/projects/CWMS/repos/cwms_database/pull-requests/319

rma-rripken committed:
Added tests.

rma-rripken committed:
Added tests.

MikeNeilson reviewed (commented):
That's looking good. I like the minor changes to organization and function names. Lot more clear.

rma-rripken committed:
standard-text-values should be empty to start.

rma-rripken committed:
IT for controller create with regular text ts.

rma-rripken committed:
Added setOffice calls to text-time-series daos.

rma-rripken committed:
Added delete integration tests.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the owning office of "
+                            + "the Text TimeSeries whose data is to be included in the response."),
+                    @OpenApiParam(name = TS_ID, required = true, description = "Specifies the ts-id of the "
```
numerical time series uses `NAME`

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                            + "the time zone of the values of the begin and end fields (unless "
+                            + "otherwise specified). If this field is not specified, "
+                            + "the default time zone of UTC shall be used."),
+                    @OpenApiParam(name = START, required = true, description = "The start of the time window"),
```
numerical time series uses `BEGIN`

adamkorynta replied:
also, is the default time window of the last 24 hours applicable here (to match numerical time series)? or should the `END` parameter also be required?

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                    @OpenApiParam(name = END, description = "The end of the time window. If specified the text associated with all times from start to end (inclusive) is deleted."),
+                    @OpenApiParam(name = VERSION_DATE, description = "The version date for the time series.  If not specified, the minimum or maximum version date (depending on p_max_version) is used."),
+                    @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
+                    @OpenApiParam(name = Controllers.MIN_ATTRIBUTE, type = Long.class, description = "The minimum attribute value to delete. If not specified, no minimum value is used."),
```
Confused by deleting an attribute here on a retrieval?

adamkorynta replied:
Now I see, it was copied from delete

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                    @OpenApiParam(name = START, required = true, description = "The start of the time window"),
+                    @OpenApiParam(name = END, description = "The end of the time window. If specified the text associated with all times from start to end (inclusive) is deleted."),
+                    @OpenApiParam(name = VERSION_DATE, description = "The version date for the time series.  If not specified, the minimum or maximum version date (depending on p_max_version) is used."),
+                    @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
```
Can we use a null `VERSION_DATE` to stand for `MAX_VERSION`?

rma-rripken replied:
yes

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                            + "otherwise specified). If this field is not specified, "
+                            + "the default time zone of UTC shall be used."),
+                    @OpenApiParam(name = START, required = true, description = "The start of the time window"),
+                    @OpenApiParam(name = END, description = "The end of the time window. If specified the text associated with all times from start to end (inclusive) is deleted."),
```
typo "deleted" -> "retrieved"

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                    @OpenApiParam(name = Controllers.MIN_ATTRIBUTE, type = Long.class, description = "The minimum attribute value to delete. If not specified, no minimum value is used."),
+                    @OpenApiParam(name = Controllers.MAX_ATTRIBUTE, type = Long.class, description = "The maximum attribute value to delete. If not specified, no maximum value is used."),
+                    @OpenApiParam(name = "mode", required = true, type = TimeSeriesTextMode.class,
+                            description = "Type of Text TimeSeries to retrieve. Options are:\n"
```
<img width="646" alt="image" src="https://github.com/USACE/cwms-data-api/assets/47677856/68be3a6e-3471-4d53-a2ce-a805f2739425">

formatting on the Mode renders funky

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+            @OpenApiParam(name = TS_ID, description = "The id of the text timeseries to be updated"),
+        },
+            queryParams = {
+                    @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
```
If we aren't using a p_version_date yet, we should leave off the `MAX_VERSION` (I'm not even sure what max_version means in the case of storage)

rma-rripken replied:
I think it means the individual rows.  it the rows don't include a version-date what happens?

adamkorynta replied:
I think that whatever the client is supplying should get stored. There shouldn't be a flag to the server specifying that some of the content is supposed to be ignored.

rma-rripken replied:
I don't think anything got ignored but I'll remove the parameter.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+        },
+            queryParams = {
+                    @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to use the maximum version date if p_version_date is not specified."),
+                    @OpenApiParam(name = REPLACE_ALL, type = Boolean.class)
```
Given `REPLACE_ALL` isn't required, what is the null behavior?

rma-rripken replied:
I added the default value to the docs

rma-rripken committed:
Added to swagger doc for methods.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                @OpenApiParam(name = VERSION_DATE, description = "The version date for the time "
+                        + "series.  If not specified, the minimum or maximum version date "
+                        + "(depending on p_max_version) is used."),
+                @OpenApiParam(name = MAX_VERSION, type = Boolean.class, description = "Whether to"
```
`MAX_VERSION` is confusing as I would think the default behavior would be true. I'd recommend leaving this parameter out and always passing in true to the pl/sql if the `VERSION_DATE` is null.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,357 @@
+                        + " the text associated with all times from start to end (inclusive) is "
+                        + "deleted."),
+                @OpenApiParam(name = VERSION_DATE, description = "The version date for the time "
+                        + "series.  If not specified, the minimum or maximum version date "
```
let's not support minimum version date at this time

rma-rripken committed:
Removing max-version from delete.

rma-rripken committed:
Using null version date instead of max_version.

rma-rripken committed:
Improving weird mode rendering.

rma-rripken committed:
making the text-timeseries path param "timeseries" to be more consistent with timeseries.

rma-rripken committed:
Improving descriptions

rma-rripken committed:
Merge remote-tracking branch 'origin/develop' into feature/text_time_series

rma-rripken committed:
Removing max-version from create and update.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -0,0 +1,380 @@
+        tags = {TAG}
+    )
+    @Override
+    public void create(@NotNull Context ctx) {
```
@MikeNeilson given the LRTS identifier updates coming down the pipeline. It it acceptable to not support storage of new LRTS text time series? Otherwise we'd need a- flag "store-as-lrts" similar to numerical time series.

I'd think skipping a "store-as-lrts" in both text and binary time series makes sense given we'll get the feature for "free" once the schema update is complete. This would mean we could turn on sessioning flags to require the new identifier and return the new identifier.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/StandardTimeSeriesTextDao.java:
```diff
@@ -0,0 +1,312 @@
+
+    public void storeRows(String officeId, String tsId, boolean maxVersion, boolean replaceAll, Collection<StandardTextTimeSeriesRow> stdRows) {
+        for (StandardTextTimeSeriesRow stdRow : stdRows) {
+            store(officeId, tsId, stdRow, maxVersion, replaceAll);
```
Parameterize in the Connection so that we aren't pulling a new connection from the pool on every row

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/RegularTimeSeriesTextDao.java:
```diff
@@ -0,0 +1,296 @@
+        // We'd have to sort the rows by textId and textValue pairs and then build a set of all the matching dates
+        // Then for each set of dates we'd call the appropriate storeTsText or storeTsTextId method.
+
+        for (RegularTextTimeSeriesRow regRow : regRows) {
```
Another place to parameterize the connection to not tax the pool

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/TextTimeSeries.java:
```diff
@@ -0,0 +1,275 @@
+// https://bitbucket.hecdev.net/projects/SC/repos/hec-monolith/browse/hec-monolith/src/main/java/hec/data/timeSeriesText/TextTimeSeries.java
+//    private ITimeSeriesDescription _timeSeriesDescription;
+    
+    private final String id;
```
rename to `name` to match numerical time series

rma-rripken replied:
this is where I was wondering if it should use something like the commented out ITimeSeriesDescription.  It would break id/name into its component parts(annoying) but let us specialize the Interval-type.   @adamkorynta  I was hoping you could weigh in on this more.  how should this handle regular, irregular, psuedo-regular etc.

adamkorynta replied:
We need the full id, the interval offset, and the location time zone. Given the LRTS interval rename migration is underway I don't think we need to do anything further as far as breakdown unless CDA started doing interval processing itself. If the timeseries description included ID, Offset, and Loc Time Zone, that would be fine. Otherwise having three separate variables seems ok to me. 

Maybe the non-graphQL-but-kinda-like-graphql updates down the line would see a catalog endpoint with just those three pieces.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/TextTimeSeries.java:
```diff
@@ -0,0 +1,275 @@
+@JsonDeserialize(builder = TextTimeSeries.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public class TextTimeSeries extends CwmsDTO {
```
Is `CwmsDTOPaginated` going to be implemented in this PR or a follow-up?

rma-rripken replied:
After.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/TextTimeSeries.java:
```diff
@@ -0,0 +1,275 @@
+@JsonDeserialize(builder = TextTimeSeries.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public class TextTimeSeries extends CwmsDTO {
```
I recommend adding the location time-zone and the interval offset similar to numerical time series for LRTS.

rma-rripken replied:
I added them to the DTO but its gonna take me a bit to figure out how to query for them.  I know the timeseries has a metadata query that fills some of that in...

rma-rripken replied:
The time-zone and interval offset will be addressed by https://github.com/USACE/cwms-data-api/issues/547

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/RegularTextTimeSeriesRow.java:
```diff
@@ -0,0 +1,223 @@
+    private final Long attribute;
+    private final String textId;
+    private final String textValue;
+    private final boolean newData;
```
Pretty sure `newData` is just used by REGI as a modified flag

rma-rripken replied:
Thank you! That has been driving me crazy

rma-rripken committed:
Parameterizing the Connection

The methods for storing data in RegularTimeSeriesTextDao and StandardTimeSeriesTextDao have been refactored. The connection setup and office setting are now performed outside the for-each loop for storing multiple rows. This optimizes the process by reducing the number of times those operations are

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dto/texttimeseries/TextTimeSeries.java:
```diff
@@ -0,0 +1,275 @@
+
+    private final NavigableMap<DateDateKey, RegularTextTimeSeriesRow> regularMap;
+    private final NavigableMap<DateDateKey, StandardTextTimeSeriesRow> stdMap;
+    private Map<String, StandardCatalog> stdCatalog = null;
```
I'm confused by the usage of the catalog here. My first thought was that this is where the standard text would match the ids to the value in order to reduce the payload sent back to the client, but the Dao looks like it just always sets the text for the standard rows.

rma-rripken replied:
Its a little convoluted.   I'll simplify and move it all into the Builder.   I still need to figure out how or when to trigger the catalog behavior - probably another flag.

MikeNeilson replied:
Given the standard text values are relatively small, the controller/dto/dao, whereever it makes sense (prob the DAO) could cache the standard values after they're retrieved the first time.

rma-rripken committed:
Removing newData.

rma-rripken committed:
Rename 'id' to 'name' in TextTimeSeries classes
Also added 'intervalOffset' and 'timeZone'.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/RegularTimeSeriesTextDao.java:
```diff
@@ -0,0 +1,307 @@
+            // 2.  make an existing textId apply at the specified times - you don't care about the current textId to textValue.
+            // This branch is if the user is trying to specify the text_Id and the text_value.
+            // We'll have to make some choices to implement this.
+            throw new UnsupportedOperationException(String.format("TextId:\"%s\" and TextValue:\"%s\" are both specified.  "
```
This winds up returning error code 500 when I think we want a 400 here

MikeNeilson replied:
Yep. Not that "unsupportedOperation" is linguistically wrong.

rma-rripken committed:
Rename 'id' to 'name'.  Removed newData field.

rma-rripken committed:
Made private

MikeNeilson commented on cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java:
```diff
@@ -226,8 +246,12 @@ public void create(@NotNull Context ctx) {
         description = "Updates a text timeseries",
         pathParams = {
-            @OpenApiParam(name = TS_ID, description = "The id of the text timeseries to be updated"),
+            @OpenApiParam(name = "timeseries", description = "The id of the text timeseries to be updated"),
```
Should we make this a constant just for the sake of consistency?

rma-rripken committed:
Moved the catalog construction into the builder.  Added tests for intervalOffset and timezone and useStandardTextCatalog

MikeNeilson reviewed (commented):
Feels like it's starting to come together.
My sense is that it's mostly nitpicky details now, next week we should probably plan, assuming what's in already works for what it is, merging in the current state so we can try to get additional feedback. Seems to me like the primary interface is "there".

rma-rripken committed:
Throw IllegalArgumentException to trigger Bad Request

rma-rripken committed:
Added a test that store can throw IllegalArgumentException.

rma-rripken committed:
Using constant from Controllers

rma-rripken committed:
Removing debug sout.

adamkorynta commented on cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/StandardTimeSeriesTextDao.java:
```diff
@@ -0,0 +1,317 @@
+        connection(dsl, connection -> {
+            setOffice(connection, officeId);
+            for (StandardTextTimeSeriesRow stdRow : stdRows) {
+                store(officeId, tsId, stdRow, maxVersion, replaceAll);
```
connection needs to be passed down into here

rma-rripken committed:
connection needed to be passed.

rma-rripken committed:
Merge pull request #546 from USACE/develop

updating to latest develop

MikeNeilson commented on cwms-data-api/src/test/resources/cwms/cda/api/spk/text_ts_create_reg.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "office-id": "SPK",
+  "id": "TsTextTestLoc.Flow.Inst.1Hour.0.raw",
```
name vs id?

rma-rripken replied:
weird that the test didn't catch that.  now I have to double check what is using that resource.

MikeNeilson commented on cwms-data-api/src/test/resources/cwms/cda/api/spk/text_ts_create_std.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "office-id": "SPK",
+  "id": "TsTextTestLoc.Flow.Inst.1Hour.0.raw",
```
name vs id?

rma-rripken replied:
should have been name.

MikeNeilson commented on cwms-data-api/src/test/resources/cwms/cda/api/spk/text_ts_update_reg.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "office-id": "SPK",
+  "id": "TsTextTestLoc.Flow.Inst.1Hour.0.raw",
```
name vs id?

MikeNeilson reviewed (changes requested):
Having a few issues actually running the tests (with the latest-dev) but that looks like a minor issue with the test setup data that I'll fix later.

However once that was solved some of the new tests didn't work due some of the input json not getting updated.

rma-rripken committed:
Updated id to name in resource files

rma-rripken committed:
Merge remote-tracking branch 'origin/develop' into feature/text_time_series

# Conflicts:
#	cwms-data-api/src/main/java/cwms/cda/ApiServlet.java
#	cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java
#	cwms-data-api/src/main/java/cwms/cda/formatters/json/JsonV2.java
#	cwms-data-api/src/test/java/cwms/cda/api/DataApiTestIT.java

MikeNeilson committed:
Tweaks to fix new and existing tests on fresh schema install.

rma-rripken committed:
Fixed merge error.

MikeNeilson committed:
Corrections to test infrastructure setup.

MikeNeilson commented:
Just updated PR to this PR, new tests still not passing. Overall structure looks good though; appears that data just isn't saving correctly but with no error message that was visible to me.

rma-rripken committed:
Widening the std delete window so that all the test data is cleaned up.

rma-rripken committed:
Have test create points and replace all.

rma-rripken committed:
Added missing resource

rma-rripken committed:
Delete twice because some tests were creating multiple versions

rma-rripken committed:
Calling DELETE_TS_TEXT directly

rma-rripken committed:
Adding @NotNull

rma-rripken committed:
Making END required.

rma-rripken committed:
Making use of Controllers constants.  Removed max-version query-params that weren't being used.  Fixed broken tests.

rma-rripken committed:
Added NotNull and Nullable annotations.

rma-rripken committed:
Using Controllers constants.  Added cleanup method. Added test assert.

rma-rripken committed:
Smaller page size b/c test harness wasn't creating enough data to trigger paging tests.

adamkorynta commented:
I pulled the latest changes in this PR and the controller integration tests pass for me.

MikeNeilson commented:
Of course now there's a conflict on one of the files.

rma-rripken committed:
Merge remote-tracking branch 'origin/develop' into feature/text_time_series

# Conflicts:
#	cwms-data-api/src/test/java/cwms/cda/data/dao/binarytimeseries/TimeSeriesBinaryDaoTestIT.java

rma-rripken committed:
Fixing imports.
